experts-api/lib/elements-client.js

```diff
@@ -65,12 +74,39 @@ export class Impersonator {
     delete options.abort_controller;
     const id = setTimeout(() => controller.abort(), timeout);
 
-    const response = await this.fetch(url, {
-      ...options,
-      signal: controller.signal
-    });
-    clearTimeout(id);
-    return response;
+    let resp;
+    try {
+      resp = await this.fetch(url, {
+        ...options,
+        signal: controller.signal
+      });
+      clearTimeout(id);
+    } catch (e) {
+      if (e.name === 'AbortError') {
+        const error=new Error('Request timed out');
+        error.status=504;
+        throw error;
+      }
+      throw e;
+    }
+
+    if (resp.status !== 204 && resp.status !== 200) {
+      let error = new Error(`CDL change propagation Error(${resp.status}):`);
+      let json=''
+      try  {
+        json=await resp.json();
+        error.message += JSON.stringify(json);
```

Comment:
## Example 3a ##
This is where we look at what client message to send along on the Error message